### DISCUSSION TITLE
How do I pass session id in when using @chain

### DISCUSSION THREAD

**Discussion by 1100111GTH** [langchain-ai/langserve#605] [Q&A]
![CleanShot 2024-04-07 at 11 20 26](https://github.com/langchain-ai/langserve/assets/154310050/55f3dd83-792e-4b24-8a0e-81dfc8f5c3b9)
![CleanShot 2024-04-07 at 11 21 14](https://github.com/langchain-ai/langserve/assets/154310050/16032fbb-e581-4a3c-b830-089299b728e9)

Langserve return runnable object, and i .stream it on the client, but can't find id

**Comment by 1100111GTH** [comment]
i am not sure what kind object type bing send with invoke two dict, so may be may source: Dict problem, but i don't know how to fix it

**Comment by eyurtsev** [comment]
I suspect that configuration isn't propagated to the `@chain`

Could you paraphrase the code to use a runnable sequence and serve that?

If you're able to share minimal reproducible code (with imports etc.), so i can copy and paste and get it working, it'll be easier for me to help

**Reply by 1100111GTH** [reply]
server code:
```python
from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder, HumanMessagePromptTemplate
from langchain_core.runnables.history import RunnableWithMessageHistory
from langchain_community.chat_message_histories.redis import RedisChatMessageHistory
from langchain_core.output_parsers import StrOutputParser
from langchain_core.runnables import chain
from typing import Any, List, Optional, Tuple, Dict

llm = ChatOpenAI(
    api_key="EMPTY", 
    base_url="http://127.0.0.1:8000/v1", 
    temperature=0,
    max_tokens=2048,
    model="qwen-14b-chat"
)

@chain
def base_answer(source: Dict):
  prompt_b = ChatPromptTemplate.from_messages(
      [
          ("system", "You are a helpful assiant", 
          MessagesPlaceholder(variable_name="history"),
          ("human", "{input}")
      ]
  )
  
  b_a = RunnableWithMessageHistory(
      prompt_b | llm, 
      RedisChatMessageHistory, 
      input_messages_key="input", 
      history_messages_key="history"
  ) | StrOutputParser()

  return b_a
```

server.py as usual...

client code ( same device for test ):
```python
from langserve import RemoteRunnable

base_answer = RemoteRunnable("http://localhost:2031/base_answer/")
base_answer.invoke({"input": "Something to say"}, {"configurable": {"session_id": "id"}})  # or .stream
```

I know the func def or not is same, but igone it. and i need streaming it on the gradio, so i really need to return the object, otherwise i will return `b_a.invoke({"input": "Something to say"}, {"configurable": {"session_id": "id"}})` this work well.

**Reply by 1100111GTH** [reply]
Whether the problem is solved or not, thank you for your help first.

**Reply by 1100111GTH** [reply]
Hi, did you find any solution？
